### ISSUE TITLE
Register Jackson modules

### ISSUE THREAD

**Issue by nipafx** [junit-pioneer/junit-pioneer#704]
One way to address #624 (JSON argument source fails with `LocalDate` property deserialization) is for Pioneer to globally register Jackson modules based on a configuration option set by the user. The option is necessary because there are different possible behaviors (e.g. only register well-known modules vs registering all modules) that don't work equally well across all test suites.

After conversations in #624 and #629, I propose the following:

* add configuration parameter `junitpioneer.jackson.modules.registration`
* implement strategies for the values `none`, `well_known`, `all`

If this proposal is accepted and implemented, a follow-up issue could add a more sophisticated option:

* for `junitpioneer.jackson.modules.registration`, add the new value `list`
* add new parameter `junitpioneer.jackson.modules.list`
* implement a strategy that if `...modules.registration` is set to `list`, the value of `...modules.list` is interpreted as a comma-separated list of Jackson modules that will be registered
* if `...modules.list` is set and `...modules.registration` is set to any other value than `list`, there should be a warning or even error

I only added this additional proposal here to foreshadow a more detailed option in case the three values `none`, `well_known`, `all` aren't flexible enough, so for this issue we can focus on implementing those three.

**Comment by nipafx** [comment]
(Not _up for grabs_ yet. Need to discuss it first.)

**Comment by meredrica** [comment]
Another approach would be to simply allow to lost the classes that should be registered. Grab the list, instanciate via reflection, crash on wrong input. This would be future proof and allows to register custom modules etc without having to go down the route of defining what well known means.

**Comment by nipafx** [comment]
That's what I meant with the follow-up proposal (under the vertical divider).

**Comment by Bukama** [comment]
I agree with @nipafx. The defensive approach with `none` as the default should be the way to go.

**Comment by meredrica** [comment]
> That's what I meant with the follow-up proposal (under the vertical divider).

I'm sorry, I have absolutely no idea why I didn't see this.

I agree that `none` should be the default.

**Comment by cwensel** [comment]
I like the idea of having the `list` escape hatch.
